src/utils/api.js

```diff
@@ -89,6 +88,29 @@ export async function initInstance(client, instance) {
     domain,
     configuration: { urls: { streaming } = {} } = {},
   } = info;
+
+  // GoToSocial requires we get the NodeInfo to identify server type
+  // spec: https://github.com/jhass/nodeinfo
+  try {
+    if (uri || domain) {
+      let urlBase = uri || `https://${domain}`;
+      const wellKnown = await (await fetch(`${urlBase}/.well-known/nodeinfo`)).json();
+      if (Array.isArray(wellKnown?.links)) {
+        const nodeInfoUrl = wellKnown.links.find(
+          (link) => typeof link.rel === 'string' &&
+          link.rel.startsWith('http://nodeinfo.diaspora.software/ns/schema/')
+        )?.href;
+        if (nodeInfoUrl && nodeInfoUrl.startsWith(urlBase)) {
+          const nodeInfo = await (await fetch(nodeInfoUrl)).json();
+          if (typeof nodeInfo?.software?.name === 'string') {
+            info.software_name = nodeInfo.software.name;
```

Comment:
Re-reading this, it seems useful to store the whole nodeinfo object somewhere else (maybe on 'nodeinfo'/'nodeinfos' key on localStorage, separate from 'instances') and not add an extra key to "info" object here to make sure there's no potential future clash.